sdk/mobile-center-crashes/src/main/java/com/microsoft/azure/mobile/crashes/ErrorAttachments.java

```diff
@@ -22,52 +23,55 @@
      *
      * @param data        binary data.
-     * @param fileName    file name to use on reports.
-     * @param contentType data MIME type.
-     * @return error attachment or null if null data is passed.
+     * @param fileName    file name to use in error attachment log.
+     * @return ErrorAttachmentLog attachment or null if null data is passed.
      */
-    public static ErrorAttachment attachmentWithBinary(byte[] data, String fileName, String contentType) {
-        return attachment(null, data, fileName, contentType);
+    public static ErrorAttachmentLog attachmentWithBinary(byte[] data, String fileName) {
+        return attachment(data, null, fileName, CONTENT_TYPE_BINARY);
     }
 
     /**
-     * Build an attachment with text and binary suitable for using in {link CrashesListener#getErrorAttachment(ErrorReport)}.
+     * Build an error attachment log with text and binary suitable for using in {link CrashesListener#getErrorAttachment(ErrorReport)}.
      *
-     * @param text        text data.
      * @param data        binary data.
-     * @param fileName    file name to use on reports for the binary data.
+     * @param text        text to attach to the attachment log.
+     * @param fileName    file name to use in error attachment log.
      * @param contentType binary data MIME type.
-     * @return error attachment or null if text and data are null.
+     * @return ErrorAttachmentLog attachment or null if text and data are null.
      */
-    public static ErrorAttachment attachment(String text, byte[] data, String fileName, String contentType) {
-        if (text == null && data == null) {
+    private static ErrorAttachmentLog attachment(byte[] data, String text, String fileName, String contentType) {
+        if ((data == null && contentType.equals(CONTENT_TYPE_BINARY)) ||
+            (text == null && contentType.equals(CONTENT_TYPE_TEXT_PLAIN))) {
             MobileCenterLog.warn(LOG_TAG, "Null content passed to attachment method, returning null");
             return null;
         }
-        ErrorAttachment attachment = new ErrorAttachment();
-        attachment.setTextAttachment(text);
-        if (data != null) {
-            ErrorBinaryAttachment binaryAttachment = new ErrorBinaryAttachment();
-            binaryAttachment.setData(data);
-            binaryAttachment.setFileName(fileName);
-            binaryAttachment.setContentType(contentType);
-            attachment.setBinaryAttachment(binaryAttachment);
-        } else if (fileName != null || contentType != null) {
-            MobileCenterLog.warn(LOG_TAG, "Binary attachment file name and content ignored as data is null");
+        if (fileName == null) {
+            MobileCenterLog.warn(LOG_TAG, "Null file name passed to attachment method, returning null");
+            return null;
+        }
+
+        String content = text;
+        if(contentType.equals(CONTENT_TYPE_BINARY)){
```

Comment:
See my comment about binary, the test should be `!contentType.startsWith("text/")`.